api/client/src/test/java/org/projectnessie/client/rest/TestResponseFilter.java

```diff
@@ -329,6 +334,37 @@ void testGood() {
     assertDoesNotThrow(() -> ResponseCheckFilter.checkResponse(new TestResponseContext(Status.OK)));
   }
 
+  @ParameterizedTest
+  @MethodSource
+  void capturingInputStream(int blockSize, byte[] source, String expected) throws Exception {
+    CapturingInputStream capturing = new CapturingInputStream(new ByteArrayInputStream(source));
+    if (blockSize == 0) {
+      while (capturing.read() >= 0) {
+        // noop
+      }
+    } else {
+      byte[] tmp = new byte[blockSize * 2];
+      while (capturing.read(tmp, 8, blockSize) >= 0) {
+        // noop
```

Comment:
Shouldn't we validate that we read the expected bytes too?

Reply:
We do that in line 354, no?

Reply:
No, on line 354 we only validate the we capture the truncated chunk correctly, but we do not validate that the normal read path returns all data without modifications.

Reply:
Isn't that testing `FilterInputStream`?

Reply:
Also, I suspect that it's working fine, since no other tests fail